Extension/src/LanguageServer/client.ts

```diff
@@ -448,6 +450,151 @@ export interface TextDocumentWillSaveParams {
     reason: vscode.TextDocumentSaveReason;
 }
 
+interface Excludes {
+    [key: string]: boolean | { when: string };
+};
+
+interface WorkspaceFolderSettingsParams {
```

Comment:
Suggestion to move any settings related interface to file "Extension/src/LanguageServer/settings.ts" to reduce code in client.ts.
`WorkspaceFolderSettingsParams`
`SettingsParams`
`InitializationOptions`

Reply:
That seems intuitive on the surface, but the purpose of `settings.ts` is for access to VS Code setting, whereas these are structures specific to the communication protocol and the native process.  Also, `InitializationOptions` does not technically related to settings, per se, other than the fact that settings are part of the initialization payload.

I will move `WorkspaceFolderSettingsParams` and `SettingsParams`, but that's not necessarily the best organization either.  :/  By the same argument, we should move the settings related serializable structures out of `serializable_objects.h` on the native side, into `settings.cpp`.  Another approach might be to create a file specifically for our protocol interfaces.

Reply:
How about create a sub file for client.ts, such as clientInterface.ts or a file that defines interfaces for the native side.

Reply:
@michelleangela That sounds good to me, but should that refactor be done later?

Reply:
A task to refactor interfaces into separate files is good.

Reply:
I actually had to move it back, for now.  It needs access to properties of client objects and is currently called in the constructor of a client object.  It requires more refactoring